scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/UploadServlet.java

```diff
@@ -149,76 +161,98 @@ private void fetchFromCatalogAndWriteResponse(String bucketName, String workflow
             LOGGER.error("Error when sending the request to catalog", e);
-        } finally {
-            try {
-                if (httpResponse != null) {
-                    httpResponse.close();
-                }
+        }
+    }
 
-            } catch (IOException e) {
-                LOGGER.error("Error when closing the response", e);
-            }
+    private void fetchJobXMLAndWriteResponse(HttpServletRequest request, HttpServletResponse response) {
+        String jobId = request.getHeader(HEADER_JOB_ID);
+        String sessionId = request.getHeader(PARAMS_SESSION_ID);
+
+        try (InputStream is = new ByteArrayInputStream(((SchedulerServiceImpl) Service.get()).getJobXML(sessionId,
+                                                                                                        jobId)
+                                                                                             .getBytes(StandardCharsets.UTF_8))) {
+            LOGGER.info("Fetching Job XML for job {}", jobId);
+            File job = File.createTempFile("job_resubmit", ".xml");
+            FileUtils.copyInputStreamToFile(is, job);
+            writeResponse(job, response);
+
+        } catch (Exception t) {
+            LOGGER.warn("Failed to download workflow xml for job " + jobId, t);
         }
     }
 
+    // Ignore Sonar's try-with-resources issue
+    // Lifecycle-management cannot be handled with try-with-resources.
+    @SuppressWarnings("squid:S2093")
     private void writeResponse(File job, HttpServletResponse response) {
         try {
-            boolean isJar = isJarFile(job);
-
-            if (!isJar) {
-                // this _loosely_ checks that the file we got is an XML file
-                try {
-                    DocumentBuilderFactory docFactory = DocumentBuilderFactory.newInstance();
-                    DocumentBuilder docBuilder = docFactory.newDocumentBuilder();
-                    Document doc = docBuilder.parse(job);
-                } catch (Throwable e) {
-                    response.getWriter().write("Job descriptor must be valid XML<br>" + e.getMessage());
-                    return;
-                }
+            if (!isJarFile(job)) {
+                checkXmlFile(job, response);
             }
-            String ret = IOUtils.toString(new FileInputStream(job), "UTF-8");
+            String ret = IOUtils.toString(new FileInputStream(job), StandardCharsets.UTF_8);
             response.getWriter().write("{ \"jobEdit\" : \"" + Base64Utils.toBase64(ret.getBytes()) + "\" }");
         } catch (IOException e) {
             LOGGER.error("Error when reading the job content", e);
             String msg = e.getMessage().replace("<", "&lt;").replace(">", "&gt;");
-            try {
-                response.getWriter().write(msg);
-            } catch (IOException ignored) {
-                // to get the error back client-side
-            }
+            writeSilently(response, msg);
```

Comment:
Funny name :) however I am not sure what it means.

Reply:
@jrochas in fact, before, writing a response message was surrounded by try-catch IOException but nothing was done in case an exception is caught (on purpose I guess) so I named it writeSilently (and don't tell me I couldn't) 😄